### PR TITLE
Implement MakeSystemProgramTransferTxData and MakeTokenProgramTransferTxData API for transferring SOL and SPL tokens

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/20698

Note: This PR is based on https://github.com/brave/brave-core/pull/12722.
## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/browser/solana_response_parser.cc [hunk 1]
```diff
@@ -185,6 +185,55 @@ bool ParseGetSignatureStatuses(
...
+  account_info.owner = *owner;
+
+  auto* data = value->FindListKey("data");
+  if (!data)
+    return false;
+  const auto& data_list = data->GetList();
+  if (data_list.size() != 2u || !data_list[0].is_string() ||
+      !data_list[1].is_string() || data_list[1].GetString() != "base64")
+    return false;
+  account_info.data = data_list[0].GetString();
```
Comment (darkdh) [inline components/brave_wallet/browser/solana_response_parser.cc:222]: I think we need to check if the data is really base64 encode here by `Base64Decode`
and add a test case for that

File: components/brave_wallet/browser/solana_tx_manager_unittest.cc [hunk 2]
```diff
@@ -322,4 +370,183 @@ TEST_F(SolanaTxManagerUnitTest, AddAndApproveTransaction) {
...
+      kSolanaTokenProgramId, std::move(account_metas), data);
+
+  std::vector<mojom::SolanaInstructionPtr> instructions;
+  instructions.push_back(mojom_transfer_instruction.Clone());
+  auto tx_data = mojom::SolanaTxData::New(
+      "", from_wallet_address, to_wallet_address, spl_token_mint_address, 0,
+      10000000, mojom::TransactionType::SolanaSPLTokenTransfer,
+      std::move(instructions));
+
+  // Owner is not token program account.
```
Comment (darkdh) [inline components/brave_wallet/browser/solana_tx_manager_unittest.cc:446]: this comment is incorrect

File: components/brave_wallet/browser/solana_response_parser.cc (hunk 1 above)
Comment (yrliou) [reply components/brave_wallet/browser/solana_response_parser.cc]: Addressed in https://github.com/brave/brave-core/pull/12729/commits/1ce4e199f80ca096af5836e593bb407bc1817b74

File: components/brave_wallet/browser/solana_tx_manager_unittest.cc (hunk 2 above)
Comment (yrliou) [reply components/brave_wallet/browser/solana_tx_manager_unittest.cc]: Fixed in https://github.com/brave/brave-core/pull/12729/commits/1ce4e199f80ca096af5836e593bb407bc1817b74, thanks

**Review by darkdh — APPROVED** [review]
lgtm now

**Comment by yrliou** [comment]
Upstream tests failed because this PR is not on master and missing fixes for that.
Upstream tests are not related to this PR in any way, so merging.

**Pull request merged**